distributed/client.py

```diff
@@ -3390,6 +3390,26 @@ def get_task_stream(self, start=None, stop=None, count=None):
         return self.sync(self.scheduler.get_task_stream, start=start,
                          stop=stop, count=count)
 
+    @gen.coroutine
+    def _register_init_func(self, function):
+        response = yield self.scheduler.register_init_func(function=dumps(function))
+        raise gen.Return(response)
```

Comment:
Instead of serializing explicitly with pickle I recommend using `yield self.scheduler.register_init_func(function=to_serialize(function))`

I think that this is all that you will need to do.  Dask should handle serialization to the worker for you.  I should probably change this in `run` as well.

Reply:
As I use `broadcast` under the hood with `run` coroutine called, this actually don't work. I probably should modify the `run` function in worker.py, but then I need to modify it in `Client.run` too. Isn't it out of scope of this PR?

Stacktrace is as follow, happens when calling `client.register_init_func`:
````
distributed.core - ERROR - 'function' object is not subscriptable
Traceback (most recent call last):
  File "/home/eh/eynardbg/git/distributed/distributed/protocol/pickle.py", line 59, in loads
    return pickle.loads(x)
TypeError: a bytes-like object is required, not 'function'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/eh/eynardbg/git/distributed/distributed/core.py", line 340, in handle_comm
    result = yield result
  File "/work/ADM/hpc/eynardbg/Outils/miniconda3/envs/dask_dev/lib/python3.5/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
  File "/work/ADM/hpc/eynardbg/Outils/miniconda3/envs/dask_dev/lib/python3.5/asyncio/futures.py", line 294, in result
    raise self._exception
  File "/work/ADM/hpc/eynardbg/Outils/miniconda3/envs/dask_dev/lib/python3.5/site-packages/tornado/gen.py", line 326, in wrapper
    yielded = next(result)
  File "/home/eh/eynardbg/git/distributed/distributed/worker.py", line 1025, in run
    function = pickle.loads(function)
  File "/home/eh/eynardbg/git/distributed/distributed/protocol/pickle.py", line 61, in loads
    logger.info("Failed to deserialize %s", x[:10000], exc_info=True)
TypeError: 'function' object is not subscriptable
distributed.core - ERROR - 'function' object is not subscriptable
Traceback (most recent call last):
  File "/home/eh/eynardbg/git/distributed/distributed/protocol/pickle.py", line 59, in loads
    return pickle.loads(x)
TypeError: a bytes-like object is required, not 'function'
````

Reply:
OK, lets not use `to_serialize` then.  You should still be able to drop loads/dumps from within the scheduler though and just store serialized bytes.

Reply:
Yep, that part with scheduler is OK.